src/aap_eda/tasks/orchestrator.py

```diff
@@ -184,4 +223,113 @@ def monitor_rulebook_processes() -> None:
+        # monitor running instances
+        LOGGER.info(f"Container engine event {event}")
+        if event.reason in ["died", "Completed", "BackoffLimitExceeded"]:
+            for process in models.RulebookProcess.objects.filter(
+                status=ActivationStatus.RUNNING, activation_pod_id=event.name
+            ):
+                try:
+                    process_parent_type = str(process.parent_type)
+                    if process_parent_type == ProcessParentType.ACTIVATION:
+                        id = process.activation_id
+                    else:
+                        id = process.event_stream_id
+                    monitor_rulebook_process(process_parent_type, id)
+                except Exception as e:
+                    LOGGER.error(f"Monitor events error {e}")
+
+
+def _get_queue_name(
+    process_parent_type: ProcessParentType, id: int
+) -> Optional[str]:
+    obj = _get_process_parent(process_parent_type, id)
+    if obj and obj.latest_instance and obj.latest_instance.node:
+        return obj.latest_instance.node.name
+
+    return "activation"
+
+
+def _get_process_parent(process_parent_type: ProcessParentType, id: int):
+    try:
+        if process_parent_type == ProcessParentType.ACTIVATION:
+            klass = Activation
+        else:
+            klass = EventStream
+        return klass.objects.get(id=id)
+    except ObjectDoesNotExist:
+        LOGGER.warning(
+            f"{process_parent_type} with {id} no longer exists, "
+            "activation manager task will not be processed",
+        )
+    return None
+
+
+def monitor_nodes() -> None:
+    """Monitor if nodes are alive.
+
+    Started by the scheduler at a regular cadence
+    """
+    LOGGER.info("Checking Worker Health")
+    activation_values = QUEUES["activation"]
+    redis = Redis(
+        host=activation_values["HOST"], port=activation_values["PORT"]
+    )
+    now = int(time.time())
+    in_use_queues = {node.name for node in models.Node.objects.all()}
+    dead_queues = set()
+    active_queues = set()
+    for worker in _Worker.all(connection=redis):
+        diff = abs(now - int(worker.last_heartbeat.timestamp()))
+        LOGGER.info(
+            f"Checking Worker {worker.name} on {worker.hostname} delay: {diff}"
+        )
+        if (
+            abs(now - int(worker.last_heartbeat.timestamp()))
+            > settings.WORKER_HEARTBEAT_DELAY
+        ):
+            LOGGER.info(
+                f"Worker {worker.name} on {worker.hostname} is not responding"
+            )
+        else:
+            for queue in worker.queues:
+                if queue.name in ["activation", "default"]:
+                    continue
+                active_queues.add(queue.name)
+
+    LOGGER.info(f"In Use Queues {in_use_queues}")
+    LOGGER.info(f"Active Queues {active_queues}")
+    for queue in in_use_queues:
+        dead_queues.add(queue)
+
+    dead_queues = in_use_queues - active_queues
```

Comment:
Here we are considering a queue dead if it has a worker dead, but it could have other workers running ok.

Reply:
Not sure about that each worker has a list of queues that they are connected to it, so if we have 2 workers
* worker1 (local_node1, activation)
* worker2(local_node1, activation)

Even if one of the dies above we will still have local_node1 and activation as the queues still available